xtc/src/cycles_wallet.rs

```diff
@@ -80,8 +84,7 @@ async fn call(args: CallCanisterArgs) -> Result<CallResult, String> {
             Ok(CallResult { r#return: x })
         }
         Err((code, msg)) => {
-            ledger.deposit(caller, args.cycles);
-
+            ledger.deposit(caller, args.cycles + deduced_fee);
```

Comment:
I don't think in case of error we should refund the fee. Imagine an attacker just create a canister that does 1 thing: traps immediately, it is probably cost him almost no cycles. However at this point our wallet already used up a lot of cycles.

Reply:
Agreed. Let's discuss it during the call.